subprojects/groovy-jsr223/src/main/java/org/codehaus/groovy/jsr223/GroovyScriptEngineImpl.java

```diff
@@ -427,7 +427,7 @@ private Object callGlobal(String name, Object[] args, ScriptContext ctx) {
     }
 
     // generate a unique name for top-level Script classes
-    private synchronized String generateScriptName() {
+    private synchronized static String generateScriptName() {
```

Comment:
Nevermind, I actually looked at what the method does and the static it increments.  This may actually be a fix. :smile:  Though doesn't static usually come before synchronized?